qa/L0_response_cache/ensemble_cache_test.py

```diff
@@ -0,0 +1,316 @@
+    def _check_single_cache_miss_success_inference(self, model_inference_stats):
+        if (
+            "count" not in model_inference_stats["cache_miss"]
+            and "count" not in model_inference_stats["cache_hit"]
+        ):
+            self.assertTrue(False, "unexpected error with response cache")
+        if (
+            "count" in model_inference_stats["cache_hit"]
+            and int(model_inference_stats["cache_hit"]["count"]) > 0
+        ):
+            self.assertTrue(False, "unexpected cache hit")
+        if int(model_inference_stats["cache_miss"]["count"]) > 1:
+            self.assertTrue(False, "unexpected multiple cache miss")
+
+    def run_ensemble(self, model):
+        model_inference_stats = self.inference_handler.get_inference_statistics(model)
+        self._check_zero_stats_baseline(model_inference_stats)
+        inference_outputs = self.inference_handler.infer()
+        self._check_valid_output(inference_outputs)
+        model_inference_stats = self.inference_handler.get_inference_statistics(model)
+        self._check_valid_stats(model_inference_stats)
+        self._check_single_cache_miss_success_inference(model_inference_stats)
+        cached_outputs = self.inference_handler.infer()
+        self._check_valid_output(cached_outputs)
+        self._check_valid_output_inference(inference_outputs, cached_outputs)
+        model_inference_stats = self.inference_handler.get_inference_statistics(model)
+        self._check_valid_stats(model_inference_stats)
+        return model_inference_stats
+
+
+class EnsembleCacheTest(tu.TestResultCollector):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.config_manager = ModelConfigManager()
+        self.inference_validator = InferenceValidator()
+        self.inference_handler = InferenceHandler()
+        self.ensemble_model = self.config_manager.get_ensemble_model()
+        self.composing_model = self.config_manager.get_composing_model()
+
+    def setup_cache_ensemble_model(self):
+        self.config_manager.enable_response_cache_ensemble_model()
+
+    def setup_decoupled_ensemble_model(self):
+        self.config_manager.enable_decoupled_ensemble_model()
+
+    def setup_decoupled_composing_model(self):
+        self.config_manager.enable_decoupled_composing_model()
+
+    def setup_cache_composing_model(self):
+        self.config_manager.enable_response_cache_composing_model()
+
+    def setup_cache_all_models(self):
+        self.config_manager.enable_response_cache_all_models()
+
+    def reset_config(self):
+        self.config_manager.reset_config_files()
+
+    def test_ensemble_top_level_cache(self):
+        model_inference_stats = self.inference_validator.run_ensemble(
+            self.ensemble_model
+        )
+        if (
+            "count" not in model_inference_stats["cache_hit"]
+            or model_inference_stats["cache_hit"]["count"] != "0"
+        ):
+            self.assertFalse(
```

Comment:
`assertFalse(False, ....)` <- not sure what was intended here, but this assert will always pass